heron/tools/common/src/python/utils/config.py

```diff
@@ -136,27 +137,38 @@ def get_heron_dir():
   its real path is '/Users/heron-user/.heron/bin/heron/tools/common/src/python/utils/config.pyc',
   the internal variable ``path`` would be '/Users/heron-user/.heron', which is the heron directory
 
+  This means the variable `go_above_dirs` below is 9.
+
   :return: root location for heron-cli.
   """
-  path = "/".join(os.path.realpath(__file__).split('/')[:-9])
+  go_above_dirs = 9
+  path = "/".join(os.path.realpath(__file__).split('/')[:-go_above_dirs])
   return normalized_class_path(path)
 
-
-################################################################################
-# Get the root of heron dir and various sub directories depending on platform
-################################################################################
-def get_heron_dir_explorer():
+def get_zipped_heron_dir():
   """
-  This will extract heron directory from .pex file.
-  From heron-cli with modification since we need to reuse cli's conf
-  :return: root location for heron-cli.
+  This will extract heron directory from .pex file,
+  with `zip_safe = False' Bazel flag added when building this .pex file
+
+  For example,
+  when __file__'s real path is
+    '/Users/heron-user/.pex/code/xxxyyy/heron/tools/common/src/python/utils/config.pyc', and
+  the internal variable ``path`` would be '/Users/heron-user/.pex/code/xxxyyy/',
+  which is the root PEX directory
+
+  This means the variable `go_above_dirs` below is 7.
+
+  Specifically designed for heron-ui
+
+  :return: root location for heron-ui.
   """
-  path_list = os.path.realpath(__file__).split('/')[:-10]
-  path_list.append(CLI_DIR)
-  path = "/".join(path_list)
+  go_above_dirs = 7
```

Comment:
I briefly mentioned it [here](https://github.com/twitter/heron/pull/1507/files/e72fe571568deba0aed16582d8367e10a48b4e2e#diff-c8ec498d0b564ed129baec8a64dceb09R159) and I am not sure how to have a better wording here.